checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleExecutor.java

```diff
@@ -61,14 +61,24 @@ public CheckstyleExecutor(CheckstyleConfiguration configuration, CheckstyleAudit
    * Execute Checkstyle and return the generated XML report.
    */
   public void execute() {
-    TimeProfiler profiler =
-            new TimeProfiler().start("Execute Checkstyle " + CheckstyleVersion.getVersion());
-    ClassLoader initialClassLoader = Thread.currentThread().getContextClassLoader();
-    Thread.currentThread().setContextClassLoader(PackageNamesLoader.class.getClassLoader());
-    URLClassLoader projectClassloader = createClassloader();
 
     Locale initialLocale = Locale.getDefault();
     Locale.setDefault(Locale.ENGLISH);
+    ClassLoader initialClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(PackageNamesLoader.class.getClassLoader());
+    try (URLClassLoader projectClassloader = createClassloader()) {
+      executeWithClassLoader(projectClassloader);
+    } catch (IOException e) {
+      throw new CheckstyleSensorException("could not close classloader", e);
```

Comment:
we can not be sure that here will be only problem with closing of classloader. you run some "executeWithClassLoader" that by name do a lot of stuff.
Why you need special exception type for this ?